### PR TITLE
test:  remove $request->uri

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
`IncomingRequest::$uri` will be protected.

- fix incorrect assertions https://github.com/codeigniter4/CodeIgniter4/commit/c42b1bfc2670d598c6431e217fc0176efcf4ace9
- refactor

**Confirm the output of `uri_string()`**

```diff
diff --git a/app/Config/Routes.php b/app/Config/Routes.php
index 0bfdce972..8650842af 100644
--- a/app/Config/Routes.php
+++ b/app/Config/Routes.php
@@ -35,7 +35,7 @@ $routes->set404Override();
 
 // We get a performance increase by specifying the default
 // route since we don't have to scan directories.
-$routes->get('/', 'Home::index');
+$routes->get('assets/(:any)', 'Home::index');
 
 /*
  * --------------------------------------------------------------------
diff --git a/app/Controllers/Home.php b/app/Controllers/Home.php
index 7f867e95f..b8cd3af74 100644
--- a/app/Controllers/Home.php
+++ b/app/Controllers/Home.php
@@ -6,6 +6,6 @@ class Home extends BaseController
 {
     public function index()
     {
-        return view('welcome_message');
+        dd(uri_string());
     }
 }
```

Navigate to http://localhost:8080/assets/image.jpg
You will see:
```
uri_string() string (16) "assets/image.jpg"
```

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: tests/system/Helpers/FormHelperTest.php [hunk 1]
```diff
@@ -195,13 +182,7 @@ public function testFormOpenWithHidden()
...
-        $config->baseURL   = '';
-        $config->indexPage = 'index.php';
-        $request           = Services::request($config);
-        $request->uri      = new URI('http://example.com/');
-
-        Services::injectMock('request', $request);
+        $this->setRequest();
 
         $before = (new Filters())->globals['before'];
         if (in_array('csrf', $before, true) || array_key_exists('csrf', $before)) {
```
Comment (MGatner) [inline tests/system/Helpers/FormHelperTest.php:188]: Everything up to here seems to be about forms; did you intend for this to be part of the PR?

File: tests/system/Helpers/URLHelper/CurrentUrlTest.php [hunk 2]
```diff
@@ -167,12 +172,11 @@ public function testUriStringNoTrailingSlashAbsolute()
...
         $this->config->baseURL = 'http://example.com';
-        $request               = Services::request($this->config);
-        $request->uri          = new URI('http://example.com/assets/image.jpg');
 
-        Services::injectMock('request', $request);
+        $uri = 'http://example.com/assets/image.jpg';
+        $this->setService($uri);
 
-        $this->assertSame('/assets/image.jpg', uri_string());
+        $this->assertSame('assets/image.jpg', uri_string());
```
Comment (MGatner) [inline tests/system/Helpers/URLHelper/CurrentUrlTest.php:179]: I'm confused why this changed just due to our fixtures. Do you know the explanation?

File: tests/system/Helpers/FormHelperTest.php (hunk 1 above)
Comment (kenjis) [reply tests/system/Helpers/FormHelperTest.php:188]: Yes, my intention is just removing the code that set `IncomingRequest::$uri` value.
That is to remove `$request->uri = new URI('http://example.com/')`.
The property will be protected.

File: tests/system/Helpers/URLHelper/CurrentUrlTest.php (hunk 2 above)
Comment (kenjis) [reply tests/system/Helpers/URLHelper/CurrentUrlTest.php:179]: Probably the test code was wrong.
The process that detects the URI is complex, and the previous test had incomplete setup.

Comment (MGatner) [reply tests/system/Helpers/URLHelper/CurrentUrlTest.php:179]: The new version looks correct so I'm gonna roll with it. URI is so unnecessarily complex 😕

**Review by MGatner — APPROVED** [review]

**Pull request merged**